hs_modelinstance/models.py

```diff
@@ -58,20 +58,50 @@ def modelProgramIdentifier(self):
 
     @classmethod
     def create(cls, **kwargs):
-        shortid = kwargs['model_name']
-        # get the MP object that matches.  Returns None if nothing is found
-        obj = ModelProgramResource.objects.filter(short_id=shortid).first()
-        metadata_obj = kwargs['content_object']
-        title = obj.title
-        return super(ExecutedBy,cls).create(model_program_fk=obj, model_name=title,content_object=metadata_obj)
+
+        # when creating a new resource version, we need to lookup the existing model_program_fk
+        if 'model_program_fk' in kwargs:
+
+            # get the foreign key id if one has been set (i.e. when creating a new version)
+            mp_short_id = kwargs['model_program_fk']
+
+            # get the MP object that matches.  Returns None if nothing is found
+            obj = ModelProgramResource.objects.filter(id=mp_short_id).first()
+
+        # when adding or changing the model_program_fk, we need to lookup the model obj based on mp_short_id
+        else:
+            # get the selected model program short id that has been submitted via the form
+            mp_short_id = kwargs['model_name']
```

Comment:
I agree. This variable is named `model_name` so that its corresponding field is displayed properly on the resource edit page (see below, `model_shortid` is ambiguous on the edit page).  I tried to find a work around for this using crispy forms but could not.  I am happy to implement a "fix" if anyone has a solution to this.

![image](https://cloud.githubusercontent.com/assets/4822372/19703243/1afe862a-9ad1-11e6-9fa1-b5dd9aee6c4d.png)